args.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package squalor
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var baseTypes = map[reflect.Type]bool{
+	reflect.TypeOf(bool(true)): true,
+	reflect.TypeOf(int(0)):     true,
+	reflect.TypeOf(int8(0)):    true,
+	reflect.TypeOf(int16(0)):   true,
+	reflect.TypeOf(int32(0)):   true,
+	reflect.TypeOf(int64(0)):   true,
+	reflect.TypeOf(uint(0)):    true,
+	reflect.TypeOf(uint8(0)):   true,
+	reflect.TypeOf(uint16(0)):  true,
+	reflect.TypeOf(uint32(0)):  true,
+	reflect.TypeOf(uint64(0)):  true,
+	reflect.TypeOf(float32(0)): true,
+	reflect.TypeOf(float64(0)): true,
+	reflect.TypeOf(string("")): true,
+}
+
+var baseKinds = map[reflect.Kind]bool{
+	reflect.Bool:    true,
+	reflect.Int:     true,
+	reflect.Int8:    true,
+	reflect.Int16:   true,
+	reflect.Int32:   true,
+	reflect.Int64:   true,
+	reflect.Uint:    true,
+	reflect.Uint8:   true,
+	reflect.Uint16:  true,
+	reflect.Uint32:  true,
+	reflect.Uint64:  true,
+	reflect.Float32: true,
+	reflect.Float64: true,
+	reflect.String:  true,
+}
+
+var kindsToBaseType = map[reflect.Kind]reflect.Type{
+	reflect.Bool:    reflect.TypeOf(bool(true)),
+	reflect.Int:     reflect.TypeOf(int(0)),
+	reflect.Int8:    reflect.TypeOf(int8(0)),
+	reflect.Int16:   reflect.TypeOf(int16(0)),
+	reflect.Int32:   reflect.TypeOf(int32(0)),
+	reflect.Int64:   reflect.TypeOf(int64(0)),
+	reflect.Uint:    reflect.TypeOf(uint(0)),
+	reflect.Uint8:   reflect.TypeOf(uint8(0)),
+	reflect.Uint16:  reflect.TypeOf(uint16(0)),
+	reflect.Uint32:  reflect.TypeOf(uint32(0)),
+	reflect.Uint64:  reflect.TypeOf(uint64(0)),
+	reflect.Float32: reflect.TypeOf(float32(0)),
+	reflect.Float64: reflect.TypeOf(float64(0)),
+	reflect.String:  reflect.TypeOf(string("")),
+}
+
+func argsConvert(from []interface{}) []interface{} {
+	to := make([]interface{}, len(from))
+	for i, arg := range from {
```

Comment:
It works with the code as is, see

https://github.com/square/squalor/commit/4a464ea0be291bfb8a2eb6aab4f0b0f54e98c104

The reason is that the type alias must be a pointer to be a `driver.Valuer`, and thus isn't considered a type alias anymore (the kind is `ptr`). Good to have a test to document this though.